### DISCUSSION TITLE
"Launch Backfill" not appearing on the Partitions tab

### DISCUSSION THREAD

**Discussion by jlloyd-widen** [dagster-io/dagster#19031] [Q&A] [answered]
I'm using the integrations with dbt (`dagster-dbt`) and Meltano (`dagster-meltano`). All of my partitioned asset jobs have a "Partitions" tab when I go to the [job details](https://docs.dagster.io/concepts/webserver/ui#job-details) page, but when I go to the Partitions tab the "Launch Backfill" is replaced by the "Materialize" button and none of my previous run jobs have logged a partition even though the partition is correctly tagged in the "Runs" tab. Here's and image of what I see:

![Screenshot 2024-01-04 at 1 46 16 PM](https://github.com/dagster-io/dagster/assets/82222659/86179673-bc7c-4106-9b81-f56ab5ab22fc)

Here's the code that sets up my Asset and Job and Schedule:
```
def meltano_asset_factory(
    tap_name: str,
    tap_streams: Mapping[str, AssetKey],
    default_job_config,
    default_env_config: dict,
) -> Tuple[list, list, list]:
    """Produce a set of assets, jobs, and schedules from basic config."""
    multi_assets = []
    jobs = []
    schedules = []

    @multi_asset(
        name=tap_name,
        resource_defs={"meltano": meltano_resource},
        compute_kind="meltano",
        group_name=tap_name,
        config_schema={
            "env": Field(
                dict,
                description="A dict of ENV objects to be injected into the cli.",
                is_required=False,
                default_value=default_env_config,
            )
        },
        outs={
            stream: AssetOut(key_prefix=[f"raw_{tap_name}"]) for stream in tap_streams
        },
    )
    def compute(context: OpExecutionContext, meltano: ConfigurableResource):
        op_name = context.op.name.replace("_", "-")
        command = f"run --force tap-{op_name} target-snowflake"
        meltano.execute_command(  # type: ignore
            f"{command}", {**os.environ, **context.op_config["env"]}, context.log
        )
        return tuple([None for _ in context.selected_output_names])

    asset_job = define_asset_job(
        name=f"{tap_name}_assets",
        selection=AssetSelection.groups(tap_name),
        config=default_job_config,
    )

    if isinstance(default_job_config, PartitionedConfig):
        partitioned_schedule = build_schedule_from_partitioned_job(
            job=asset_job,
            hour_of_day=1,
            minute_of_hour=0,
        )
        schedules.append(partitioned_schedule)

    multi_assets.append(compute)
    jobs.append(asset_job)

    return multi_assets, jobs, schedules


def generate_default_op_envs(d: datetime):
    """Generate a configurable default set of env vars for the tap."""
    return {
        "TARGET_SNOWFLAKE_DEFAULT_TARGET_SCHEMA": "salesforce",
        "TAP_SALESFORCE_START_DATE": d.strftime("%Y-%m-%dT00:00:00Z"),
    }


@daily_partitioned_config(
    start_date=datetime(2023, 3, 1),
    timezone="UTC",
    fmt="%Y-%m-%d",
    end_offset=0,
)
def daily_config(start: datetime, _end: datetime):
    """Configure each daily partition run."""
    return {
        "resources": {"meltano": {"config": {"project_dir": f"{os.getcwd()}/meltano"}}},
        "ops": {"salesforce": {"config": {"env": generate_default_op_envs(start)}}},
    }


salesforce_assets, salesforce_jobs, salesforce_schedules = meltano_asset_factory(
    tap_name="salesforce",
    tap_streams=get_asset_keys_by_output_name_for_source(
        dbt_assets=[edp_dbt_assets], source_name="raw_salesforce"
    ),
    default_job_config=daily_config,
    default_env_config=generate_default_op_envs(yesterday),
)
```

Is this a bug, or have I configured things incorrectly?

**Comment by clairelin135** [comment]
Hi @jlloyd-widen. 

This is odd... I tried to reproduce with a condensed version of your code snippet but I got the expected UI:
<img width="1957" alt="image" src="https://github.com/dagster-io/dagster/assets/29110579/142670fc-6bd6-4a68-bdf7-06ccae195de1">

Code snippet:
```python
@daily_partitioned_config(
    start_date=datetime(2023, 3, 1),
    timezone="UTC",
    fmt="%Y-%m-%d",
    end_offset=0,
)
def daily_config(start: datetime, _end: datetime):
    """Configure each daily partition run."""
    return {
        "resources": {"meltano": {"config": {"project_dir": f"blah/meltano"}}},
        "ops": {"salesforce": {"config": {"env": "blah"}}},
    }


asset_job = define_asset_job(
    name="assets",
    config=daily_config,
)

partitioned_schedule = build_schedule_from_partitioned_job(
    job=asset_job,
    hour_of_day=1,
    minute_of_hour=0,
)

defs = Definitions([], jobs=[asset_job], schedules=[partitioned_schedule])
```

Can you try updating your dagster version? If that doesn't work, can you try with my code snippet and see if that loads correctly?

**Reply by jlloyd-widen** [reply]
updating to 1.5.14 didn't improve anything, nor did your simplified code snippet 😢

**Reply by clairelin135** [reply]
Would you mind sharing a screenshot of what you see with the simplified code snippet?

**Reply by jlloyd-widen** [reply]
It's literally the same as the screenshot above, nothing changed but the asset count (which makes sense)
![Screenshot 2024-01-05 at 1 51 15 PM](https://github.com/dagster-io/dagster/assets/82222659/b45d18e8-81e2-40b4-9fda-7abd8679e7f7)

**Comment by clairelin135** [comment, marked as answer]
@jlloyd-widen I was able to reproduce this issue.

Looks like there is a rough edge when using partitioned config with assets. Partitioned config applies a partitions definition at the job level, but not at the per asset level, causing the UI to believe the asset is unpartitioned.

You can work around this by defining a partitions def that matches the partitioned config partitions def on the asset level, i.e.:
```python

@daily_partitioned_config(
    start_date=datetime(2023, 3, 1),
    timezone="UTC",
    fmt="%Y-%m-%d",
    end_offset=0,
)
def daily_config(start: datetime, _end: datetime):
    """Configure each daily partition run."""
    return {
        "ops": {"my_asset": {"config": {"env": "blah"}}},
    }


@asset(
    partitions_def=DailyPartitionsDefinition("2023-03-01")
)
def my_asset():
    ...
```
Then the UI will behave as expected.

For now, I'll file an issue so we can track this in our backlog.

**Reply by clairelin135** [reply]
Issue: https://github.com/dagster-io/dagster/issues/19076

**Reply by jlloyd-widen** [reply]
Well, the workaround almost worked. When I alter the factory to include the `partitions_def` like you have it in your example, I also have to rework a bit of the `define_asset_job` call, but in the end does alter the UI a bit in a positive direction:

<img width="1460" alt="Screenshot 2024-01-09 at 10 58 58 AM" src="https://github.com/dagster-io/dagster/assets/82222659/e1c70d55-2fdd-4ca7-bca4-2ce6f4e359cb">

So I can now see a count of the partitions, but I still don't see the `Launch Backfill` text. Then when I actually run a partition I get the error described in #17380 . Instead of seeing anything on the partitions tab.
